dev/com.ibm.ws.session/src/com/ibm/ws/session/SessionData.java

```diff
@@ -10,151 +10,51 @@
  *******************************************************************************/
 package com.ibm.ws.session;
 
-import java.io.IOException;
-import java.io.ObjectInput;
-import java.io.ObjectOutput;
 import java.util.Enumeration;
-import java.util.HashMap;
-import java.util.LinkedList;
 import java.util.Vector;
 import java.util.logging.Level;
 
 import javax.servlet.ServletContext;
-import javax.servlet.http.HttpServletRequest;
 
-import com.ibm.websphere.security.WebSphereRuntimePermission;
-import com.ibm.websphere.servlet.session.IBMApplicationSession;
 import com.ibm.ws.session.http.HttpSessionImpl;
 import com.ibm.ws.session.utils.LoggingUtil;
 import com.ibm.ws.util.ArrayEnumeration;
-import com.ibm.wsspi.servlet.session.IBMSessionExt;
 import com.ibm.wsspi.session.ISession;
 
-/*
+/**
  * The WAS Http Session adaptation.  Extends the core HttpSessionImpl and adds WAS-specific function
+ * <p>
+ * Servlet 6 update: 
+ *      - common methods/APIs are moved to AbstractSessionData
+ *      - Keep some methods (including supportive methods) here to override the HttpSessionImpl
  */
-public class SessionData extends HttpSessionImpl implements IBMSessionExt {
+public class SessionData extends HttpSessionImpl {
 
     private static final long serialVersionUID = -76305717244905946L;
     protected HttpSessionFacade _httpSessionFacade;
-    protected static final String SECURITY_PROP_NAME = "com_ibm_ejs_security_httpsession_info";
+    private static final String methodClassName = "SessionData";
 
-    protected boolean _hasSecurityInfo = true; // used in
+    //protected boolean _hasSecurityInfo = true; // used in
                                                // getAttributeNames/getValueNames
     // we have to assume true and we'll set to false
     // in getValueNames if the sec prop isn't found
 
-    protected LinkedList lockList;
-    protected HashMap locks = null;
-
-    protected String pathInfoForAppSession = null;
     String sipCookieInfo = null;
```

Comment:
sipCookieInfo is in the abstract class.  No need to duplicate it here.

Reply:
cleanup